src/server/stream_family.cc

```diff
@@ -2542,6 +2574,34 @@ void StreamFamily::XRangeGeneric(CmdArgList args, bool is_rev, ConnectionContext
   return (*cntx)->SendError(result.status());
 }
 
+void StreamFamily::XAck(CmdArgList args, ConnectionContext* cntx) {
+  string_view key = ArgS(args, 0);
+  string_view group = ArgS(args, 1);
+  args.remove_prefix(2);
+  absl::InlinedVector<streamID, 8> ids(args.size());
+
+  for (size_t i = 0; i < args.size(); ++i) {
+    ParsedStreamId parsed_id;
+    string_view str_id = ArgS(args, i);
+    if (!ParseID(str_id, true, 0, &parsed_id)) {
+      return (*cntx)->SendError(kInvalidStreamId, kSyntaxErrType);
+    }
+    ids[i] = parsed_id.val;
+  }
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    return OpAck(t->GetOpArgs(shard), key, group, absl::Span{ids.data(), ids.size()});
+  };
+
+  OpResult<uint32_t> result = cntx->transaction->ScheduleSingleHopT(std::move(cb));
```

Comment:
`std::move` doesn't do anything here :) I will do a full review later today (At the gym ATM)

Reply:
fixed.